scalecube-configuration/src/test/java/io/scalecube/configuration/repository/InMemoryConfigurationRepository.java

```diff
@@ -31,74 +38,124 @@
 
   @Override
   public Mono<Document> read(String tenant, String repository, String key, Integer version) {
-    return get(new Repository(tenant, repository), key);
+    return Mono.justOrEmpty(getRepositoryKey(new Repository(tenant, repository), key, version));
   }
 
   @Override
   public Flux<Document> readAll(String tenant, String repository, Integer version) {
-    return entries(new Repository(tenant, repository));
+    Set<Document> values =
+        getRepository(new Repository(tenant, repository)).values().stream()
+            .map(
+                vl ->
+                    version == null
+                        ? vl.get(vl.size() - 1)
+                        : version <= vl.size() ? vl.get(version - 1) : null)
+            .filter(v -> v != null)
+            .collect(Collectors.toSet());
```

Comment:
Is it possible don't collect it again?

Reply:
It's local testing and performance is very high